macosx/Controller.m

```diff
@@ -853,6 +859,41 @@ - (void) applicationWillTerminate: (NSNotification *) notification
     tr_sessionClose(fLib);
 }
 
+- (void) setupTouchBarIfReady
+{
+    if (touchbar != nil)
+    {
+        [touchbar setDefaultItemIdentifiers: @[createTouchBarItem.identifier,
+                                               openFileTouchBarItem.identifier,
+                                               removeTouchBarItem.identifier,
+                                               NSTouchBarItemIdentifierFixedSpaceSmall,
+                                               pauseAllTouchBarItem.identifier,
+                                               resumeAllTouchBarItem.identifier,
+                                               NSTouchBarItemIdentifierFlexibleSpace,
+                                               shareTouchBarButton.identifier
+                                               ]];
+        [touchbar setCustomizationAllowedItemIdentifiers: @[createTouchBarItem.identifier,
+                                                            openFileTouchBarItem.identifier,
+                                                            pauseAllTouchBarItem.identifier,
+                                                            removeTouchBarItem.identifier,
+                                                            resumeAllTouchBarItem.identifier,
+                                                            speedLimitTouchBarItem.identifier,
+                                                            cleanupTouchBarItem.identifier,
+                                                            quickLookTouchBarItem.identifier,
+                                                            showInfoTouchBarItem.identifier,
+                                                            NSTouchBarItemIdentifierFlexibleSpace,
+                                                            NSTouchBarItemIdentifierFixedSpaceLarge,
+                                                            NSTouchBarItemIdentifierFixedSpaceSmall,
+                                                            shareTouchBarButton.identifier
+                                                            ]];
+    }
+    else
+    {
+        [NSTimer scheduledTimerWithTimeInterval:0.1 target:self selector: @selector(setupTouchBarIfReady) userInfo:nil repeats:NO];
```

Comment:
If I understand this correctly, this method will be called indefinitely if touchbar is never wired. Is this the case for systems lacking the touchbar? I've seen other apps use
```
[NSApp respondsToSelector:@selector(setAutomaticCustomizeTouchBarMenuItemEnabled:)]
```
to test for presence of the touchbar.